fdbserver/include/fdbserver/MoveKeys.actor.h

```diff
@@ -43,17 +43,31 @@ struct MoveKeysLock {
 };
 
 class DDEnabledState {
+	enum Value {
+		ENABLED, // DD is enabled
+		SNAPSHOT, // disabled for snapshot
+		BLOB_RESTORE_PREPARING // disabled for hybrid restore
+	};
+
 	// in-memory flag to disable DD
-	bool ddEnabled = true;
 	UID ddEnabledStatusUID;
+	Value stateValue = ENABLED;
 
 public:
-	// checks if DD is enabled/disable transiently
-	bool isDDEnabled() const;
+	bool sameId(const UID&) const;
+
+	bool isEnabled() const;
+
+	bool isBlobRestorePreparing() const;
 
-	// transiently enable(true) or disable(false) the DD. If the process
+	// transiently enable the DD. If the process
 	// restarts, the state will be forgotten.
-	bool setDDEnabled(bool status, UID snapUID);
+	bool setDDEnabled(UID requesterId);
+
+	// transiently disable the DD
+	bool setDDSnapshot(UID requesterId);
+
+	bool setDDRestorePreparing(UID requesterId);
```

Comment:
I think it'll be good to document here that these functions will return false and not change the state if ...

Reply:
Yeah. And renaming the 3 functions to `tryXXX` can indicate the semantics better

Reply:
Yes, `try...` naming would also make it a lot clearer that this will not always set the status and that the caller should look at the return value to determine if the set worked.